src/view_control.cpp

```diff
@@ -637,7 +637,21 @@ void View::DrawOctave(
 
     if ((spanningType == SPANNING_START_END) || (spanningType == SPANNING_END)) {
         if (octave->HasEndid()) {
-            x2 += (m_doc->GetGlyphWidth(SMUFL_E0A2_noteheadWhole, staff->m_drawingStaffSize, false) / 2);
```

Comment:
Has you tried using `LayerElement::GetContentX2`? Maybe you need to check that the bounding box is set, though.